tt_llk_wormhole_b0/common/inc/sfpu/ckernel_sfpu_welfords.h

```diff
@@ -0,0 +1,188 @@
+    //Next cycle cannot read from LREG4 See tt-isa-documentation
+
+    /*var_{N+1}temp = 1/(N+1)*/
+    float inv_n_plus_1 = 1.0/((float)N+1.0);
+    uint32_t bits = __builtin_bit_cast(std::uint32_t, inv_n_plus_1);
+    uint16_t high16_bits = static_cast<uint16_t>(bits >> 16);
+    uint16_t low16_bits  = static_cast<uint16_t>(bits & 0xFFFF);
+    /*var_{N+1}temp = 1/(N+1) usage loads high 16 bits*/
+    TT_SFPLOADI(p_sfpu::LREG7, 8, high16_bits);
+
+    /*mean_{N+1}temp = 1 * (InputLREG + (-mean))*/
+    TTI_SFPADD(p_sfpu::LCONST_1/*LREG10 = <1>*/, InputLREG, p_sfpu::LREG4, p_sfpu::LREG6,0);
+    //Next cycle cannot read from LREG6 See tt-isa-documentation
+
+    /*mean_{N} = -1 * ((-1) *mean_{N})*/
+    TTI_SFPMULI(negative_one_BF16, p_sfpu::LREG4, 0);
+    //Next cycle cannot read from LREG4 See tt-isa-documentation
+
+    /*var_{N+1}temp = 1/(N+1) usage loads low 16 bits*/
+    TT_SFPLOADI(p_sfpu::LREG7, 10, low16_bits);
+
+    /*mean_{N+1} = ((mean_{N+1} = (InputLREG-mean) * (1/N+1)) + mean_{N}*/
+    TTI_SFPMAD(p_sfpu::LREG6, p_sfpu::LREG7, p_sfpu::LREG4, p_sfpu::LREG6, 0);
+    //Next cycle cannot read from LREG6 See tt-isa-documentation
+
+    //mean calculation end
+    //
+    //var calculation start
+
+
+    //var_{N+1} = var_{N} + ...
+    //...(1/N+1) * (((x_{N+1} - mean_{N}) * (x_{N+1} - mean_{N+1})) - var_{N})
+
+    /*mean_{N} = (Input_LREG - mean_{N})*/
+    TTI_SFPMAD(p_sfpu::LREG11, p_sfpu::LREG4, InputLREG, p_sfpu::LREG4, 0);
+    //Next cycle cannot read from LREG4 See tt-isa-documentation
+
+    /*inputLREG temp = (InputLREG + (-mean_{N+1}))*/
+    TTI_SFPMAD(p_sfpu::LREG11/*LREG11 = <-1>*/,p_sfpu::LREG6, InputLREG, InputLREG,0);
+    //Next cycle cannot read from InputLREG See tt-isa-documentation
+
+    //var_N = -var_N
+    TTI_SFPMULI(negative_one_BF16, p_sfpu::LREG5, 0);
+    //Next cycle cannot read from LREG5 See tt-isa-documentation
+    TTI_SFPNOP;
+
+    /*InputLREG temp= (mean_{N} = (InputLREG-mean_{N}) * InputLREG =(InputLREG - mean_{N+1})) + var_{N}*/
+    TTI_SFPMAD(p_sfpu::LREG4, InputLREG, p_sfpu::LREG5, InputLREG, 0);
+    //Next cycle cannot read from InputLREG See tt-isa-documentation
+
+    //var_N = -(-var_N)
+    TTI_SFPMULI(negative_one_BF16, p_sfpu::LREG5, 0);
+    TTI_SFPNOP;
+
+    //updates the var
+    /*var_{N} = (InputLREG = (see above) * (1/N+1)) + var_{N} = ()*/
+    TTI_SFPMAD(InputLREG, p_sfpu::LREG7, p_sfpu::LREG5, p_sfpu::LREG5, 0);
+    //Next cycle cannot read from LREG5 See tt-isa-documentation
+
+    //keep it simple please change
+    /*mean_{N} = -1 * ((-1) *mean_{N+1})*/
+    TTI_SFPMUL(p_sfpu::LCONST_1 /*LREG11 = <-1>*/, p_sfpu::LREG6, p_sfpu::LCONST_0, p_sfpu::LREG4, 0);
+    //Next cycle cannot read from LREG4 See tt-isa-documentation
+    TTI_SFPNOP;
+
+    //var calculation end
+    //Now past_mean (LREG4) is population
+    //Now past_var (LREG5) is population
+}
+
+void Welfords_Main(uint32_t &N, uint32_t &endN, uint32_t &last_run) {
```

Comment:
```suggestion
void Welfords_Main(uint32_t &N, const uint32_t endN) {
```
`last_run` seems to be unused in the method. Consider removing it.
`endN` isn't changed anywhere, pass it as a const. Although it doesn't make any difference from the compiler's point of view, it sends a clear message to the reader about constness of the variable.